File: Python/game.py
import random

def create_grid(rows, cols):
    return [[random.randint(0,1) for _ in range(cols)] for _ in range(rows)]

def count_alive_neighbors(grid, row, col):
    count = 0
    for i in range(-1, 2):
        for j in range(-1, 2):
            if i == 0 and j == 0:
                continue
            r, c = row + i, col + j
            if (0 <= r < len(grid)) and (0 <= c < len(grid[0])):
                count += grid[r][c]
    return count

def step(grid):
    new_grid = [[0 for _ in range(len(grid[0]))] for _ in range(len(grid))]
    for row in range(len(grid)):
        for col in range(len(grid[0])):
            alive_neighbors = count_alive_neighbors(grid, row, col)
            if grid[row][col] == 1:
                if alive_neighbors < 2 or alive_neighbors > 3:
                    new_grid[row][col] = 0
                else:
                    new_grid[row][col] = 1
            else:
                if alive_neighbors == 3:
                    new_grid[row][col] = 1
    return new_grid

grid = create_grid(10, 10)
for _ in range(10):
    grid = step(grid)
    for row in grid:
        print(row)
    print()

File: C++/game.cpp
#include <iostream>
#include <vector>
#include <ctime>


using namespace std;

const int ROWS = 30;
const int COLS = 30;

vector<vector<int>> grid(ROWS, vector<int>(COLS));
vector<vector<int>> newGrid(ROWS, vector<int>(COLS));

void initializeGrid() {
    // Initialize the grid with random values
    for (int row = 0; row < ROWS; row++) {
        for (int col = 0; col < COLS; col++) {
            grid[row][col] = rand() % 2;
        }
    }
}

void displayGrid() {
    // Display the grid
    for (int row = 0; row < ROWS; row++) {
        for (int col = 0; col < COLS; col++) {
            cout << grid[row][col] << " ";
        }
        cout << endl;
    }
}

int countNeighbors(int row, int col) {
    int count = 0;
    for (int i = row - 1; i <= row + 1; i++) {
        for (int j = col - 1; j <= col + 1; j++) {
            if (i >= 0 && i < ROWS && j >= 0 && j < COLS) {
                if (grid[i][j] == 1 && !(i == row && j == col)) {
                    count++;
                }
            }
        }
    }
    return count;
}

void updateGrid() {
    // Update the grid for the next generation
    for (int row = 0; row < ROWS; row++) {
        for (int col = 0; col < COLS; col++) {
            int neighbors = countNeighbors(row, col);
            if (grid[row][col] == 1) {
                if (neighbors < 2 || neighbors > 3) {
                    newGrid[row][col] = 0;
                }
                else {
                    newGrid[row][col] = 1;
                }
            }
            else {
                if (neighbors == 3) {
                    newGrid[row][col] = 1;
                }
                else {
                    newGrid[row][col] = 0;
                }
            }
        }
    }
    grid = newGrid;
}

int main() {
    srand(time(0));
    initializeGrid();
    while (true) {
        system("cls");
        displayGrid();
        updateGrid();
        _sleep(500);
    }
    return 0;
}
